libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c

```diff
@@ -2289,7 +2289,11 @@ TEST( MQTT_Unit_API, GetIncomingMQTTPacketTypeAndLengthChecks )
 
     /* Test with invalid remaining length. */
     buffer[ 0 ] = 0x20; /* CONN ACK */
-    buffer[ 1 ] = 0xFF; /* Remaining length. */
+    /* To generate invalid remaining length response,
+     * three bytes need to have MSB (or continuation bit, 0x80) set */
+    buffer[ 1 ] = 0xFF;
+    buffer[ 2 ] = 0xFF; 
```

Comment:
This change also seems unrelated to removing `goto`, so please call it out in the commit message.

Reply:
It is in the commit message , "PR also contains a unit test fix."